### PR TITLE
Update documentation to support InnerProduct with k-NN Lucene Engine

### PR DESCRIPTION
**Pull request by navneet1v**
### Description
Update documentation to support InnerProduct with k-NN Lucene Engine

### Issues Resolved
K-NN issue: https://github.com/opensearch-project/k-NN/issues/1550

Closes https://github.com/opensearch-project/documentation-website/issues/6704

### Checklist
- [x] By submitting this pull request, I confirm that my contribution is made under the terms of the Apache 2.0 license and subject to the [Developers Certificate of Origin](https://github.com/opensearch-project/OpenSearch/blob/main/CONTRIBUTING.md#developer-certificate-of-origin).
For more information on following Developer Certificate of Origin and signing off your commits, please check [here](https://github.com/opensearch-project/OpenSearch/blob/main/CONTRIBUTING.md#developer-certificate-of-origin).

### CODE REVIEW

**Comment by navneet1v** [comment]
Screenshot 2024-03-17 at 11 21 01 PM

Adding the screen shot for the table

File: _search-plugins/knn/knn-index.md
Comment (kolchfa-aws) [inline _search-plugins/knn/knn-index.md:94]:
```suggestion
`hnsw` | false | l2, cosinesimil, innerproduct (supported in OpenSearch version 2.13 and later) | Hierarchical proximity graph approach to Approximate k-NN search.
```

File: _search-plugins/knn/knn-score-script.md
Comment (kolchfa-aws) [inline _search-plugins/knn/knn-score-script.md:323]:
```suggestion
        \[ \text{If} d > 0, score = d + 1 \] \[\text{If} d \le 0\] \[score = {1 \over 1 + (-1 &middot; d) }\]
```

**Comment by navneet1v** [comment]
> Thank you, @navneet1v! A couple of formatting and style changes.

sure thanks Let me just update the PR

File: _search-plugins/knn/approximate-knn.md
Comment (jmazanec15) [inline _search-plugins/knn/approximate-knn.md:292]: From user perspective, score is in effect the negative dot product. Do we need to specify difference here if user won't know about it?

Comment (navneet1v) [reply _search-plugins/knn/approximate-knn.md:292]: I was bit of in split mind here. My thinking is we tell users that our dot product for Faiss is `-1 * dot product` which comes out of our Faiss Engine class. But that is not same for Lucene Engine. Hence I thought its better to put that distinction.

But open to suggestion as in both the cases the score are same.

Comment (jmazanec15) [reply _search-plugins/knn/approximate-knn.md:292]: I see. Yes, then I think this is fine.

Comment (navneet1v) [reply _search-plugins/knn/approximate-knn.md:292]: I removed the script score formula based on the discuss with @jmazanec15 that script score uses same for all the engines.

**Review by jmazanec15 — APPROVED** [review]

**Comment by navneet1v** [comment]
@kolchfa-aws can I get review from your side on this PR

**Review by kolchfa-aws — APPROVED** [review]
LGTM. Thanks, @navneet1v!

**Comment by navneet1v** [comment]
@kolchfa-aws can you merge the pr?

**Comment by kolchfa-aws** [comment]
Will merge once the editorial review is complete.

File: _search-plugins/knn/knn-index.md
Comment (natebower) [inline _search-plugins/knn/knn-index.md:94]:
```suggestion
`hnsw` | false | l2, cosinesimil, innerproduct (supported in OpenSearch 2.13 and later) | Hierarchical proximity graph approach to approximate k-NN search.
```

Comment (natebower) [inline _search-plugins/knn/knn-index.md:94]: It looks like we are inconsistent in our capitalization of "approximate" here: https://opensearch.org/docs/latest/search-plugins/knn/approximate-knn/.

**Review by natebower — COMMENTED** [review]
@navneet1v @kolchfa-aws Just one comment/change combo for you. Thanks!

File: _search-plugins/knn/knn-index.md
Comment (navneet1v) [reply _search-plugins/knn/knn-index.md]: @natebower fixed all the places where there was issue in the file.

Comment (kolchfa-aws) [reply _search-plugins/knn/knn-index.md]: Thank you, @navneet1v! Ready to merge.

**Pull request merged**
